apps/auth/nextjs-ssr/src/hooks/auth.ts

```diff
@@ -0,0 +1,29 @@
+import { cookies } from "next/headers";
+import { createCrossmint, CrossmintAuth } from "@crossmint/server-sdk";
+
+export async function getAuthSession(refreshRoute?: string) {
+    const cookieStore = await cookies();
+    const jwt = cookieStore.get("crossmint-jwt")?.value;
+    const refreshToken = cookieStore.get("crossmint-refresh-token")?.value;
+
+    if (!refreshToken) {
+        return;
+    }
+
+    try {
+        const crossmint = createCrossmint({
+            apiKey: process.env.SERVER_CROSSMINT_API_KEY || "",
+        });
+        const crossmintAuth = CrossmintAuth.from(crossmint, {
+            refreshRoute,
+        });
+
+        const session = await crossmintAuth.getSession({
```

Comment:
just overall meta comment: it seems weird that u need both the refresh and the jwt to get the session, or is that wrong? i think with just the jwt you should be able to

Reply:
I see how the naming is confusing yeah, but really the core token is the refreshToken as the jwt is almost always expired.